googlemaps/places.py

```diff
@@ -0,0 +1,203 @@
+                     the time the query is sent.
+    :type open_now: bool
+
+    :param types: Restricts the results to places matching at least one of the
+                  specified types.
+    :type types: string or list of strings
+
+    :param page_token: Token from a previous search that when provided will
+                       returns the next page of results for the same search.
+    :type page_token: string
+
+    :rtype: result dict with the following keys:
+            status: status code
+            results: list of places
+            html_attributions: set of attributions which must be displayed
+            next_page_token: token for retrieving the next page of results
+
+    """
+    params = {}
+
+    if query:
+        params["query"] = query
+    if location:
+        params["location"] = convert.latlng(location)
+    if radius:
+        params["radius"] = radius
+    if language:
+        params["language"] = language
+    if min_price:
+        params["minprice"] = min_price
+    if max_price:
+        params["maxprice"] = max_price
+    if open_now:
+        params["opennow"] = "true"
+    if page_token:
+        params["pagetoken"] = page_token
+
+    return client._get("/maps/api/place/textsearch/json", params)
+
+
+def place(client, place_id, language=None):
+    """
+    Comprehensive details for an individual place.
+
+    :param place_id: A textual identifier that uniquely identifies a place,
+                     returned from a Places search.
+    :type place_id: string
+
+    :param language: The language in which to return results.
+    :type langauge: string
+
+    :rtype: result dict with the following keys:
+            status: status code
+            result: dict containing place details
+            html_attributions: set of attributions which must be displayed
+
+    """
+    params = {"placeid": place_id}
+    if language:
+        params["language"] = language
+    return client._get("/maps/api/place/details/json", params)
+
+
+def photo(client, photo_reference, max_width=None, max_height=None):
+    """
+    Downloads a photo from the Places API.
+
+    :param photo_reference: A string identifier that uniquely identifies a
+      photo, as provided by either a Places search or Places detail request.
+    :type photo_reference: string
+
+    :param max_width: Specifies the maximum desired width, in pixels.
+    :type max_width: int
+
+    :param max_height: Specifies the maximum desired height, in pixels.
+    :type max_height: int
+
+    :rtype: iterator containing the raw image data, which typically can be
+      used to save an image file locally, eg:
+
+    ```
+    f = open(local_filename, 'wb')
+    for chunk in client.photo(photo_reference, max_width=100):
+        if chunk:
+            f.write(chunk)
+    f.close()
+    ```
+
+    """
+
+    if not (max_width or max_height):
+        raise ValueError("a max_width or max_height arg is required")
```

Comment:
It throws a 400 error without any meaningful message, and returns a default image (an icon that looks broken).

I figure it's worth trapping if it at worst it saves a HTTP request and at best saves someone a broken app.

What do you think?